core/src/main/scala-3/medeia/generic/GenericProductDecoder.scala

```diff
@@ -14,10 +14,12 @@ import medeia.decoder.BsonDecoder
 import shapeless3.deriving.*
 
 object GenericProductDecoder {
-  private case class Accumulator(labels: IndexedSeq[String], errors: Chain[BsonDecoderError]) {
-    def next(): Accumulator = this.copy(labels.tail)
-    def next(error: BsonDecoderError) = this.copy(labels.tail, errors :+ error)
-    def next(errors: NonEmptyChain[BsonDecoderError]) = this.copy(labels.tail, this.errors ++ errors.toChain)
+  private case class Accumulator(labels: IndexedSeq[String], errors: Option[NonEmptyChain[BsonDecoderError]]) {
```

Comment:
Replaced `Chain[BsonDecoderError]` with `Option[NonEmptyChain[BsonDecoderError]]` which is basically the same but nicer to handle in the cases below